src/extensions/default/CodeFolding/foldhelpers/handlebarsFold.js

```diff
@@ -0,0 +1,174 @@
+/**
+ * Fold range finder for handlebars/mustache template type files.
+ * @author Patrick Oladimeji
+ * @date 14/08/2016 22:04:21
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+    var CodeMirror = brackets.getModule("thirdparty/CodeMirror/lib/codemirror"),
+        endOfLineSpaceRegex = /\s$/;
+
+    /**
+     * Utility function for scanning the text in a document until a certain condition is met
+     * @param {object}  cm  The code mirror object representing the document
+     * @param {string}  startCh  The start character position for the scan operation
+     * @param {number}  startLine The start line position for the scan operation
+     * @param {function} condition A predicate function that takes in the text seen so far and returns true if the scanning process should be halted
+     * @returns {{from:CodeMirror.Pos, to: CodeMirror.Pos, string: string}} An object representing the range of text scanned.
+     */
+    function scanTextUntil(cm, startCh, startLine, condition) {
+        var line = cm.getLine(startLine),
+            seen = "",
+            characterIndex = startCh,
+            currentLine = startLine,
+            range;
+        while (currentLine <= cm.lastLine()) {
+            if (line.length === 0) {
+                characterIndex = 0;
+                line = cm.getLine(++currentLine);
+            } else {
+                seen = seen.concat(line[characterIndex] || "");
+                if (condition(seen)) {
+                    range = {
+                        from: {ch: startCh, line: startLine},
+                        to: {ch: characterIndex, line: currentLine},
+                        string: seen
+                    };
+                    return range;
+                } else if (characterIndex >= line.length) {
+                    seen = seen.concat(cm.lineSeparator());
+                    if (condition(seen)) {
+                        range = {
+                            from: {ch: startCh, line: startLine},
+                            to: {ch: characterIndex, line: currentLine},
+                            string: seen
+                        };
+                        return range;
+                    }
+                    characterIndex = 0;
+                    line = cm.getLine(++currentLine);
+                } else {
+                    ++characterIndex;
+                }
+            }
+        }
+    }
+
+    function endTag(seen) {
+        return endOfLineSpaceRegex.test(seen) || seen.endsWith("}");
```

Comment:
You have to use the `StringUtils.endsWith` counterpart.
According to [MDN](https://developer.mozilla.org/it/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith) Chrome support it since version 41 and Linux is at 29, and IE doesn't support it and we will break the in browser approach.